agent/socket.go

```diff
@@ -126,35 +147,60 @@ func getPodFromCallbackEvent(cb *slack.InteractionCallback) (string, string, err
 	return split[0], split[1], nil
 }
 
-func getTimeFromCallbackEvent(cb *slack.InteractionCallback, baseTime time.Time) (time.Time, error) {
-	if _, ok := cb.BlockActionState.Values[extendBlockID]; !ok {
-		return time.Time{}, fmt.Errorf("failed to get value from callback: block_id(%s) is not specified", extendBlockID)
-	}
-	if _, ok := cb.BlockActionState.Values[extendBlockID][pickerActionID]; !ok {
-		return time.Time{}, fmt.Errorf("failed to get value from callback: action_id(%s) is not specified", pickerActionID)
+func (s *Server) extendPod(ctx context.Context, channel, namespace, pod string) error {
+	po, err := s.clientset.CoreV1().Pods(namespace).Get(ctx, pod, metav1.GetOptions{})
+	if err != nil {
+		s.log.Error(err, "failed to get pod info",
+			"name", pod,
+			"namespace", namespace,
+		)
+		return err
 	}
 
-	selectedTime := cb.BlockActionState.Values[extendBlockID][pickerActionID].SelectedTime
-	split := strings.Split(selectedTime, ":")
-	if len(split) != 2 {
-		return time.Time{}, fmt.Errorf("invalid time format: %s", selectedTime)
-	}
-	hh, err := strconv.Atoi(split[0])
+	status, err := s.runnerClient.GetStatus(ctx, po.Status.PodIP)
 	if err != nil {
-		return time.Time{}, fmt.Errorf("invalid time format: %s", selectedTime)
+		s.log.Error(err, "failed to get runner status",
+			"name", pod,
+			"namespace", namespace,
+		)
+		return err
 	}
-	mm, err := strconv.Atoi(split[1])
+	var tm time.Time
+	if status.DeletionTime != nil {
+		tm = *status.DeletionTime
+	} else {
+		tm = time.Now().UTC() // We now expect DeletionTime is not nil at this point. This line is a fail safe.
+		s.log.Info("deletion time was not set beforehand",
+			"name", pod,
+			"namespace", namespace,
+		)
+	}
+	tm = tm.Add(time.Hour * extendDurationHours)
+	limit := time.Now().UTC().Add(time.Hour * extendLimitHours)
```

Comment:
Is it intentional to use current time as the basis?
It can be extended indefinitely by shifting the time.

Reply:
It is intentional.

> re-extention

Nothing problem because users push extend button since they want to exntend. The limit is just a fool-proof to prevent unintentionally extend very long period.